test/unit/throttle_enumerator_test.rb

```diff
@@ -102,6 +102,7 @@ def each_iteration(_record, _params); end
       enqueued = ActiveJob::Base.queue_adapter.enqueued_jobs
       assert_equal 1, enqueued.size
       assert_equal 0, enqueued.first.fetch("cursor_position")
+      assert_equal 1, enqueued.first.fetch("times_interrupted")
```

Comment:
This is the core issue I wanted to solve here, and this assertion fails on main.